### ISSUE TITLE
File integrity checksum errors during image push

### ISSUE THREAD

**Issue by wittlesouth** [kubernetes/kubernetes#58588]
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG REPORT

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**: Pushing an image from a remote client (which is a MacOS machine), image push failed for larger layers with messages like:

file integrity checksum failed for "usr/lib/x86_64-linux-gnu/gconv/libJISX0213.so"

**What you expected to happen**: Successful ability to push image to registry

**How to reproduce it (as minimally and precisely as possible)**:

1) Build a Ubuntu 16.04 based image with Docker for Mac 17.12.0-ce-mac49 (21995) running on OS X 10.13.2..
2) Label for a remote Kubernetes node running the registry.
3) Push it to the registry

**Anything else we need to know?**:

Assuming the environment matters here.

**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.1", GitCommit:"3a1c9449a956b6026f075fa3134ff92f7d55f812", GitTreeState:"clean", BuildDate:"2018-01-04T11:52:23Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", GitCommit:"5fa2db2bd46ac79e5e00a4e6ed24191080aa463b", GitTreeState:"clean", BuildDate:"2018-01-18T09:42:01Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration:
Running on an internal Kubernetes cluster build on Ubuntu 16.04, Intel NUC7i5BNK
- OS (e.g. from /etc/os-release):
eric@nuc1:~$ more /etc/os-release
NAME="Ubuntu"
VERSION="16.04.3 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.3 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
- Kernel (e.g. `uname -a`):
Linux nuc1 4.13.0-26-generic #29~16.04.2-Ubuntu SMP Tue Jan 9 22:00:44 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
Not sure what you're looking for here, build the Kubernetes cluster with kubadm, and installed the service per the instructions on this site.
- Others:
NFS mount for the persistent volume is from a Synology DS415+ running DSM 6.1.4-15217 Update 5

**Comment by wittlesouth** [comment]
Additional note, tried uploading an image from the Linux host running the Kubernetes cluster, it still fails, this time I got this from the client:

received unexpected HTTP status: 500 Internal Server Error

Logs from the pod show:

192.168.5.10 - - [21/Jan/2018:19:27:40 +0000] "PATCH /v2/eric/olsnet2_server/blobs/uploads/57af9745-6694-4f51-9758-83956ae1836d?_state=3MUsLYGhEvWl7ytaBTMwUTpH-av9oOTIdqJ_hjHyU417Ik5hbWUiOiJlcmljL29sc25ldDJfc2VydmVyIiwiVVVJRCI6IjU3YWY5NzQ1LTY2OTQtNGY1MS05NzU4LTgzOTU2YWUxODM2ZCIsIk9mZnNldCI6MCwiU3RhcnRlZEF0IjoiMjAxOC0wMS0yMVQxOToyNzozMC45OTUxNDg0NjNaIn0%3D HTTP/1.1" 500 193 "-" "docker/1.13.1 go/go1.6.2 git-commit/092cba3 kernel/4.13.0-26-generic os/linux arch/amd64 UpstreamClient(Docker-Client/1.13.1 \x5C(linux\x5C))" "-"
192.168.5.10 - - [21/Jan/2018:19:27:41 +0000] "PATCH /v2/eric/olsnet2_server/blobs/uploads/b8b690b7-4fca-4e4a-b116-40b205fe0021?_state=8JrE0G4Hv_VX4BDmG_R2DcMc3I0F9JCQuAm3jZAkqHx7Ik5hbWUiOiJlcmljL29sc25ldDJfc2VydmVyIiwiVVVJRCI6ImI4YjY5MGI3LTRmY2EtNGU0YS1iMTE2LTQwYjIwNWZlMDAyMSIsIk9mZnNldCI6MCwiU3RhcnRlZEF0IjoiMjAxOC0wMS0yMVQxOToyNzozMC4zODk0OTg5MTlaIn0%3D HTTP/1.1" 500 193 "-" "docker/1.13.1 go/go1.6.2 git-commit/092cba3 kernel/4.13.0-26-generic os/linux arch/amd64 UpstreamClient(Docker-Client/1.13.1 \x5C(linux\x5C))" "-"

**Comment by wittlesouth** [comment]
I agree the report is more suitable for the registry on top of kubernetes project. I found the Kubernetes registry add-on at this GitHub page: 

https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/registry

When I clicked on the "issues" link for that page, it took me to the core Kubernetes issue section where I posted this bug. If the bugs for the Kubernetes registry add-on are not tracked here, can you point me to where they are tracked?

**Comment by ericchiang** [comment]
Ah. Based on my reading of the docs, the addons directory is deprecated https://kubernetes.io/docs/concepts/cluster-administration/addons/#legacy-add-ons

That registry directory hasn't had much traffic, and there's no OWNERS file either. Maybe issues like https://github.com/moby/moby/issues/33991 would be better?
